[ICE]: `no entry for node id`

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_ast_lowering/src/lib.rs:670:55: 'no entry for node id: `NodeId(638)`'', 'thread 'rustc' ($TID) panicked at compiler/rustc_ast_lowering/src/lib.rs:670:55: 'no entry for node id: `NodeId(638)`''
File: /tmp/im/a.rs
-->

auto-reduced (treereduce-rust):
````rust
impl Trait {
    reuse None::<&()>;
}

````

original:
````rust
#![feature(fn_delegation)]
use std::mem::type_info::Trait;

impl Trait {
    reuse None::<&()>;
}

fn main() {}

````

Version information
````
rustc 1.97.0-nightly (3e353d735 2026-05-08)
binary: rustc
commit-hash: 3e353d7353b0cd6fa4eefd5c882457cc2708184c
commit-date: 2026-05-08
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.4

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/3e353d7353b0cd6fa4eefd5c882457cc2708184c/compiler/rustc_ast_lowering/src/lib.rs#L664-L676

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0425]: cannot find type `Trait` in this scope
 --> /tmp/icemaker_global_tempdir.BoZwu6yIhSbv/rustc_testrunner_tmpdir_reporting.GcV6o0pQ8QxL/mvce.rs:1:6
  |
1 | impl Trait {
  |      ^^^^^ not found in this scope
  |
help: consider importing this struct
  |
1 + use std::mem::type_info::Trait;
  |

error[E0423]: expected function, found unit variant `None`
 --> /tmp/icemaker_global_tempdir.BoZwu6yIhSbv/rustc_testrunner_tmpdir_reporting.GcV6o0pQ8QxL/mvce.rs:2:11
  |
2 |     reuse None::<&()>;
  |           ^^^^^^^^^^^ not a function

error[E0658]: functions delegation is not yet fully implemented
 --> /tmp/icemaker_global_tempdir.BoZwu6yIhSbv/rustc_testrunner_tmpdir_reporting.GcV6o0pQ8QxL/mvce.rs:2:5
  |
2 |     reuse None::<&()>;
  |     ^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #118212 <https://github.com/rust-lang/rust/issues/118212> for more information
  = help: add `#![feature(fn_delegation)]` to the crate attributes to enable
  = note: this compiler was built on 2026-05-08; consider upgrading it if it is out of date


thread 'rustc' (694310) panicked at /rustc-dev/3e353d7353b0cd6fa4eefd5c882457cc2708184c/compiler/rustc_ast_lowering/src/lib.rs:670:55:
no entry for node id: `NodeId(17)`
stack backtrace:
   0:     0x7f690b874239 - <<std[db65628f2330fd11]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[b3c617e8375f9bd0]::fmt::Display>::fmt
   1:     0x7f690be1b788 - core[b3c617e8375f9bd0]::fmt::write
   2:     0x7f690b88ac66 - <std[db65628f2330fd11]::sys::stdio::unix::Stderr as std[db65628f2330fd11]::io::Write>::write_fmt
   3:     0x7f690b84a2de - std[db65628f2330fd11]::panicking::default_hook::{closure#0}
   4:     0x7f690b8678f3 - std[db65628f2330fd11]::panicking::default_hook
   5:     0x7f690a7544d1 - std[db65628f2330fd11]::panicking::update_hook::<alloc[b10a1c065189447]::boxed::Box<rustc_driver_impl[1f444ccaa450aabc]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7f690b867bd2 - std[db65628f2330fd11]::panicking::panic_with_hook
   7:     0x7f690b84a3a2 - std[db65628f2330fd11]::panicking::panic_handler::{closure#0}
   8:     0x7f690b8410b9 - std[db65628f2330fd11]::sys::backtrace::__rust_end_short_backtrace::<std[db65628f2330fd11]::panicking::panic_handler::{closure#0}, !>
   9:     0x7f690b84be0d - __rustc[87b1f669d989e0ab]::rust_begin_unwind
  10:     0x7f690892c50c - core[b3c617e8375f9bd0]::panicking::panic_fmt
  11:     0x7f690c505d13 - <rustc_ast_lowering[33be4fe4c65770bd]::LoweringContext>::new_named_lifetime
  12:     0x7f690c506ed8 - <rustc_ast_lowering[33be4fe4c65770bd]::LoweringContext>::lower_ty
  13:     0x7f690c50b258 - <rustc_ast_lowering[33be4fe4c65770bd]::LoweringContext>::lower_qpath
  14:     0x7f690a27a0e2 - <rustc_ast_lowering[33be4fe4c65770bd]::LoweringContext>::generate_delegation_error
  15:     0x7f690a275087 - <rustc_ast_lowering[33be4fe4c65770bd]::LoweringContext>::lower_delegation
  16:     0x7f690c6ff514 - <rustc_ast_lowering[33be4fe4c65770bd]::item::ItemLowerer>::lower_node
  17:     0x7f690a274b2b - rustc_ast_lowering[33be4fe4c65770bd]::lower_delayed_owner
  18:     0x7f690bf7162c - rustc_query_impl[3b9ecabc2cca1a8f]::execution::try_execute_query::<rustc_data_structures[4a3bb46d707d6722]::vec_cache::VecCache<rustc_span[749aae40fdac821d]::def_id::LocalDefId, rustc_middle[4a2cff3451d2bad5]::query::erase::ErasedData<[u8; 0usize]>, rustc_middle[4a2cff3451d2bad5]::dep_graph::graph::DepNodeIndex>, false>
  19:     0x7f690b29e31b - rustc_query_impl[3b9ecabc2cca1a8f]::query_impl::lower_delayed_owner::execute_query_non_incr::__rust_end_short_backtrace
  20:     0x7f690d1449c0 - rustc_middle[4a2cff3451d2bad5]::hir::map::hir_crate_items
  21:     0x7f690cf85119 - rustc_query_impl[3b9ecabc2cca1a8f]::query_impl::hir_crate_items::invoke_provider_fn::__rust_begin_short_backtrace
  22:     0x7f690cf815f1 - rustc_query_impl[3b9ecabc2cca1a8f]::execution::try_execute_query::<rustc_middle[4a2cff3451d2bad5]::query::caches::SingleCache<rustc_middle[4a2cff3451d2bad5]::query::erase::ErasedData<[u8; 8usize]>>, false>
  23:     0x7f690cf8050f - rustc_query_impl[3b9ecabc2cca1a8f]::query_impl::hir_crate_items::execute_query_non_incr::__rust_end_short_backtrace
  24:     0x7f690bf6ecd8 - rustc_interface[232b63f46aa35c0e]::passes::analysis
  25:     0x7f690cf822c9 - rustc_query_impl[3b9ecabc2cca1a8f]::execution::try_execute_query::<rustc_middle[4a2cff3451d2bad5]::query::caches::SingleCache<rustc_middle[4a2cff3451d2bad5]::query::erase::ErasedData<[u8; 0usize]>>, false>
  26:     0x7f690cf81f33 - rustc_query_impl[3b9ecabc2cca1a8f]::query_impl::analysis::execute_query_non_incr::__rust_end_short_backtrace
  27:     0x7f690cfc765b - rustc_interface[232b63f46aa35c0e]::interface::run_compiler::<(), rustc_driver_impl[1f444ccaa450aabc]::run_compiler::{closure#0}>::{closure#1}
  28:     0x7f690cf75c3e - std[db65628f2330fd11]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[232b63f46aa35c0e]::util::run_in_thread_with_globals<rustc_interface[232b63f46aa35c0e]::util::run_in_thread_pool_with_globals<rustc_interface[232b63f46aa35c0e]::interface::run_compiler<(), rustc_driver_impl[1f444ccaa450aabc]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  29:     0x7f690cf7632d - <std[db65628f2330fd11]::thread::lifecycle::spawn_unchecked<rustc_interface[232b63f46aa35c0e]::util::run_in_thread_with_globals<rustc_interface[232b63f46aa35c0e]::util::run_in_thread_pool_with_globals<rustc_interface[232b63f46aa35c0e]::interface::run_compiler<(), rustc_driver_impl[1f444ccaa450aabc]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[b3c617e8375f9bd0]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  30:     0x7f690cf7712c - <std[db65628f2330fd11]::sys::thread::unix::Thread>::new::thread_start
  31:     0x7f6906aa597a - <unknown>
  32:     0x7f6906b292bc - <unknown>
  33:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.97.0-nightly (3e353d735 2026-05-08) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [lower_delayed_owner] lowering the delayed AST owner `<impl at /tmp/icemaker_global_tempdir.BoZwu6yIhSbv/rustc_testrunner_tmpdir_reporting.GcV6o0pQ8QxL/mvce.rs:1:1: 1:11>::None`
#1 [hir_crate_items] getting HIR crate items
#2 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 3 previous errors

Some errors have detailed explanations: E0423, E0425, E0658.
For more information about an error, try `rustc --explain E0423`.

```

</p>
</details>

<!--
query stack:
#0 [lower_delayed_owner] lowering the delayed AST owner `<impl at /tmp/icemaker_global_tempdir.BoZwu6yIhSbv/rustc_testrunner_tmpdir_reporting.GcV6o0pQ8QxL/mvce.rs:1:1: 1:11>::None`
#1 [hir_crate_items] getting HIR crate items
#2 [analysis] running analysis passes on crate `mvce`

-->
@rustbot label +F-fn_delegation

--- a/compiler/rustc_resolve/src/late.rs
+++ b/compiler/rustc_resolve/src/late.rs
@@ -3905,7 +3905,11 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
             self.visit_ty(&qself.ty);
         }
 
-        self.visit_path(&delegation.path);
+        // Create lifetimes not with `LifetimeRibKind::Generics` but with `LifetimeRibKind::Elided`,
+        // as we are not processing generic params but generic args in a future function call (#156342).
+        self.with_lifetime_rib(LifetimeRibKind::Elided(LifetimeRes::Infer), |this| {
+            this.visit_path(&delegation.path);
+        });
 
         self.r.delegation_infos.insert(
             self.r.current_owner.def_id,

--- a/tests/ui/delegation/generics/generics-gen-args-errors.rs
+++ b/tests/ui/delegation/generics/generics-gen-args-errors.rs
@@ -114,8 +114,7 @@ mod test_3 {
     //~| ERROR: method takes 1 lifetime argument but 0 lifetime arguments were supplied
 
     reuse Trait::<Trait, Clone, _, 'static, dyn Send, _>::foo::<1, 2, 3, _, 6> as bar7;
-    //~^ ERROR: missing lifetime specifiers [E0106]
-    //~| ERROR: trait takes 3 lifetime arguments but 1 lifetime argument was supplied
+    //~^ ERROR: trait takes 3 lifetime arguments but 1 lifetime argument was supplied
     //~| ERROR: trait takes 2 generic arguments but 5 generic arguments were supplied
     //~| ERROR: method takes 2 generic arguments but 5 generic arguments were supplied
     //~| ERROR: method takes 1 lifetime argument but 0 lifetime arguments were supplied

--- a/tests/ui/delegation/generics/generics-gen-args-errors.stderr
+++ b/tests/ui/delegation/generics/generics-gen-args-errors.stderr
@@ -59,17 +59,6 @@ help: consider introducing lifetime `'a` here
 LL |     reuse foo'a, ::<"asdasd", asd, "askdn", 'static, 'a> as bar7;
    |              +++
 
-error[E0106]: missing lifetime specifiers
-  --> $DIR/generics-gen-args-errors.rs:116:19
-   |
-LL |     reuse Trait::<Trait, Clone, _, 'static, dyn Send, _>::foo::<1, 2, 3, _, 6> as bar7;
-   |                   ^^^^^ expected 3 lifetime parameters
-   |
-help: consider introducing a named lifetime parameter
-   |
-LL |     reuse Trait::<Trait<'a, 'a, 'a>, Clone, _, 'static, dyn Send, _>::foo'a, ::<1, 2, 3, _, 6> as bar7;
-   |                        ++++++++++++                                      +++
-
 error[E0423]: expected value, found struct `String`
   --> $DIR/generics-gen-args-errors.rs:13:33
    |
@@ -697,7 +686,7 @@ error[E0747]: constant provided when a type was expected
 LL |     reuse foo::<{}, {}, {}> as bar8;
    |                 ^^
 
-error: aborting due to 58 previous errors
+error: aborting due to 57 previous errors
 
-Some errors have detailed explanations: E0106, E0107, E0121, E0261, E0401, E0423, E0425, E0747.
-For more information about an error, try `rustc --explain E0106`.
+Some errors have detailed explanations: E0107, E0121, E0261, E0401, E0423, E0425, E0747.
+For more information about an error, try `rustc --explain E0107`.

--- a/tests/ui/delegation/wrong-lifetime-rib-ice-156342.rs
+++ b/tests/ui/delegation/wrong-lifetime-rib-ice-156342.rs
@@ -1,0 +1,17 @@
+#![feature(fn_delegation)]
+#![feature(type_info)]
+
+use std::mem::type_info::Trait;
+
+impl Trait {
+//~^ ERROR: cannot define inherent `impl` for a type outside of the crate where the type is defined
+    reuse None::<&()>;
+    //~^ ERROR: expected function, found unit variant `None`
+}
+
+fn foo<T>() {}
+
+reuse foo::<&&&&&&&&&&()> as foo1;
+reuse foo::<&std::borrow::Cow<'_, &()>> as foo2;
+
+fn main() {}

--- a/tests/ui/delegation/wrong-lifetime-rib-ice-156342.stderr
+++ b/tests/ui/delegation/wrong-lifetime-rib-ice-156342.stderr
@@ -1,0 +1,19 @@
+error[E0423]: expected function, found unit variant `None`
+  --> $DIR/wrong-lifetime-rib-ice-156342.rs:8:11
+   |
+LL |     reuse None::<&()>;
+   |           ^^^^^^^^^^^ not a function
+
+error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined
+  --> $DIR/wrong-lifetime-rib-ice-156342.rs:6:1
+   |
+LL | impl Trait {
+   | ^^^^^^^^^^ impl for type defined outside of crate
+   |
+   = help: consider defining a trait and implementing it for the type or using a newtype wrapper like `struct MyType(ExternalType);` and implement it
+   = note: for more details about the orphan rules, see <https://doc.rust-lang.org/reference/items/implementations.html?highlight=orphan#orphan-rules>
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0116, E0423.
+For more information about an error, try `rustc --explain E0116`.

